core/trino-main/src/main/java/io/trino/sql/planner/plan/AggregationNode.java

```diff
@@ -104,6 +112,23 @@ public AggregationNode(
         outputs.addAll(groupingSets.getGroupingKeys());
         hashSymbol.ifPresent(outputs::add);
         outputs.addAll(aggregations.keySet());
+        if (step.isOutputPartial() && !groupingSets.getGroupingKeys().isEmpty()) {
+            // add mask channels
```

Comment:
> How is this related to this change?

Because you only output `aggregation input` symbols and `mask` as part of PA. You don't output `oderingScheme` symbols or `filter` symbol. Hence you need checks that these are not needed

Reply:
I guess we either need to skip the aggregations with `oderingScheme` and `filter` (with validations on top) or support it in the final step.

Reply:
Look at the classes I pointed. They don't support intermediate aggregation

Reply:
discussed offline